proofs/sectorbuilder/testing/interface_test.go

```diff
@@ -357,4 +357,38 @@ func TestSectorBuilder(t *testing.T) {
 			t.Fatalf("timed out waiting for seal to complete")
 		}
 	})
+
+	t.Run("PIP generation and verification", func(t *testing.T) {
+		h := NewBuilder(t).Build()
+		defer h.Close()
+
+		inputBytes := RequireRandomBytes(t, h.MaxBytesPerSector.Uint64())
+		ref, size, reader, err := h.CreateAddPieceArgs(inputBytes)
+		require.NoError(t, err)
+
+		sectorID, err := h.SectorBuilder.AddPiece(context.Background(), ref, size, reader)
+		require.NoError(t, err)
+
+		timeout := time.After(MaxTimeToSealASector)
+
+		select {
+		case val := <-h.SectorBuilder.SectorSealResults():
+			require.NoError(t, val.SealingErr)
+			require.NotNil(t, val.SealingResult)
+			require.Equal(t, sectorID, val.SealingResult.SectorID)
+			require.Equal(t, 1, len(val.SealingResult.Pieces), "too many pieces")
```

Comment:
nit: maybe the case where the number of pieces is 0 is impossible, but if not the error message would cause some confusion in that case.